harness/libraries/influx_handler.py

```diff
@@ -0,0 +1,315 @@
+            if int(r.status_code) >= 400:
+                self.logger.doWarningLogging(f"Influx request failed, status_code = {r.status_code}, text = {r.text}, reason = {r.reason}.")
+                return []
+        except requests.exceptions.ConnectionError as e:
+            self.logger.doErrorLogging("InfluxDB is not reachable. Request not sent.")
+            self.logger.doErrorLogging(str(e))
+            return []
+        except Exception as e:
+            self.logger.doErrorLogging(f"Failed to send to {influx_url}:")
+            self.logger.doErrorLogging(str(e))
+            return []
+        resp = r.json()
+        ret_data = self._parse_query_result_from_json(resp)
+        return ret_data
+
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+    #                                                                 @
+    # End of public methods.                                          @
+    #                                                                 @
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+    #                                                                 @
+    # Private methods                                                 @
+    #                                                                 @
+    #@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+    def _check_url_endpoint(self, url, read=True, v2_api=True):
+        """
+            Checks the endpoint of the URL
+            Parameters:
+                url: URL to check
+                read: if True, it is a read-request (ie, query). else, it is a write-request
+                v2_api: if True, checks that endpoint is consistent with InfluxDB v2 HTTP API. Else, checks for v1 compatibility
+        """
+        url_parsed = urlparse(url)
+        endpoint = url_parsed.path
+        target_endpoint = endpoint
+        if read and v2_api:
+            # Then the endpoint should be /api/v2/query?org=accept
+            if not endpoint.startswith('/api/v2/query'):
+                target_endpoint = f'/api/v2/query'
+                self.logger.doWarningLogging(f'Found bad endpoint for v2 read request. Got: {endpoint}, expected {target_endpoint}.')
+        elif read and (not v2_api):
+            # Then the endpoint should be /query?org=accept
+            if not endpoint.startswith('/query'):
+                target_endpoint = f'/query'
+                self.logger.doWarningLogging(f'Found bad endpoint for v1 read request. Got: {endpoint}, expected {target_endpoint}.')
+        elif not read:
+            # All write queries have the same endpoint -- /api/v2/write
+            if not endpoint.startswith('/api/v2/write'):
+                target_endpoint = f'/api/v2/write'
+                self.logger.doWarningLogging(f'Found bad endpoint for v2 write request. Got: {endpoint}, expected {target_endpoint}.')
```

Comment:
This line enforces that all POST requests are `/api/v2/write` endpoint, which is invalid for influxDB versions <2.0